src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -70,9 +71,14 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
 
                 foreach (var projectReference in libraries.Where(e => e.Type == LibraryType.Project || e.Type == LibraryType.ExternalProject))
                 {
+                    var projectLibrary = assetsFile.Libraries.Single(a => a.Name == projectReference.Name && a.Version == projectReference.Version);
+                    var id = string.Equals(Path.GetFileNameWithoutExtension(projectLibrary.MSBuildProject), projectReference.Name, StringComparison.OrdinalIgnoreCase)
```

Comment:
for consistency should this be platform aware? 
I know it doesn't change much now. 

Do what you think is best in this case :)